rewrite-groovy/src/main/java/org/openrewrite/groovy/GroovyParserVisitor.java

```diff
@@ -2309,7 +2301,25 @@ private int sourceLengthOfString(ConstantExpression expr) {
             int delimiterLength = getDelimiterLength();
             return delimiterLength + value.length() + delimiterLength;
         }
-        return lengthAccordingToAst(expr);
+        int lengthAccordingToAst = lengthAccordingToAst(expr);
+        // subtract any parentheses that were included with lengthAccordingToAst
+        Integer insideParenthesesLevel = getInsideParenthesesLevel(expr);
+        if (insideParenthesesLevel != null) {
+            return lengthAccordingToAst - insideParenthesesLevel * 2;
+        }
+
+        return lengthAccordingToAst;
+    }
+
+    private static @Nullable Integer getInsideParenthesesLevel(ASTNode node) {
+        Object rawIpl = node.getNodeMetaData("_INSIDE_PARENTHESES_LEVEL");
+        if (rawIpl instanceof AtomicInteger) {
+            // On Java 11 and newer _INSIDE_PARENTHESES_LEVEL is an AtomicInteger
+            return ((AtomicInteger) rawIpl).get();
+        } else {
+            // On Java 8 _INSIDE_PARENTHESES_LEVEL is a regular Integer
+            return (Integer) rawIpl;
```

Comment:
I'm not seeing any evidence that this is ever triggered on Groovy 2.5; might need to change how we detect and handle parentheses for that version.